tools/isobuild/bundler.js

```diff
@@ -792,6 +806,33 @@ class Target {
                 this.nodeModulesDirectories[unibuild.nodeModulesPath] = nmd;
               }
               f.nodeModulesDirectory = nmd;
+
+              if (!archinfo.matches(this.arch, unibuild.arch)) {
+                // The unibuild we're trying to include doesn't work for the
+                // bundle target (eg, os.osx.x86_64 instead of os.linux.x86_64)!
+                // Hopefully this is because we specially enabled the feature
+                // that leads to this.
+                if (!this.providePackageJSONForUnavailableBinaryDeps) {
+                  throw Error("mismatched arch without special feature enabled "
```

Comment:
OK, I think it would be good to document that in a comment -  sometimes it's hard to differentiate between errors that should never happen and errors that are user-facing.